src/Image.js

```diff
@@ -0,0 +1,95 @@
+import { useElementSize } from '@kaliber/use-element-size'
+import imageUrlBuilder from '@sanity/image-url'
+
+const SIZES = [320, 480, 720, 1024, 1440, 1920, 2760, 3600]
+
+export function Image({ sanityConfig, image, layoutClassName = undefined, imgProps = {} }) {
+  const { ref: sizeRef, size: { width = 0 } } = useElementSize()
+  const { src, srcSet } = useSrcSet({ config: sanityConfig, image })
+
+  return (
+    <img
+      {...imgProps}
+      ref={sizeRef}
+      className={layoutClassName}
+      sizes={Math.max(1, width) + 'px'}
+      alt={image.alt}
+      width={image.asset.metadata.dimensions.width}
+      height={image.asset.metadata.dimensions.height}
+      {...{ src, srcSet }}
+    />
+  )
+}
```

Comment:
Ik zou hier nog een base class voor maken waarbij je de properties die verschillen meegeeft. Het basis mechanisme van `useSrcSet` en `sizes` is hetzelfde en ik denk dat dat duidelijk maakt wat nou precies het verschil tussen de 3 is (ik kan het mis hebben).